detectable/src/test/java/com/synopsys/integration/detectable/detectables/dart/unit/PubDepsParserTest.java

```diff
@@ -0,0 +1,44 @@
+package com.synopsys.integration.detectable.detectables.dart.unit;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+
+import com.synopsys.integration.bdio.graph.DependencyGraph;
+import com.synopsys.integration.bdio.model.Forge;
+import com.synopsys.integration.bdio.model.externalid.ExternalIdFactory;
+import com.synopsys.integration.detectable.detectables.dart.pubdep.PubDepsParser;
+import com.synopsys.integration.detectable.util.graph.NameVersionGraphAssert;
+
+public class PubDepsParserTest {
+    @Test
+    public void test() throws IOException, URISyntaxException {
+        File pubDeps = new File(this.getClass().getClassLoader().getResource("detectables/unit/dart/pubDeps.txt").toURI());
```

Comment:
This test should be moved to the` functional` test package as it uses resources.
`FunctionalTestFiles::asListOfString(relativePath)` will make this a lot cleaner.